pkg/gather/collect/discoverer.go

```diff
@@ -0,0 +1,191 @@
+package collect
+
+import (
+	"fmt"
+	"slices"
+
+	"github.com/scylladb/scylla-operator/pkg/helpers"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/discovery"
+	"k8s.io/klog/v2"
+)
+
+// alwaysIgnoredResources holds resources that should never be collected.
+var alwaysIgnoredResources = []schema.GroupKind{
+	// Built-in Secrets.
+	{
+		Group: "",
+		Kind:  "Secret",
+	},
+	// Bitnami SealedSecrets.
+	{
+		Group: "bitnami.com",
+		Kind:  "SealedSecret",
+	},
```

Comment:
Q: is this API enough of a thing to justify including it here? First time I'm seeing it.